### ISSUE TITLE
运行之后卡住

### ISSUE THREAD

**Issue by BeSkyer** [OpenRLHF/OpenRLHF#864]
运行之后只会占两张卡，且会卡住不动（等了一个小时也没有变化

(LLMRayActor pid=1404966) INFO 03-13 01:59:49 model_runner.py:1562] Graph capturing finished in 16 secs, took 0.20 GiB
(LLMRayActor pid=1404966) INFO 03-13 01:59:49 llm_engine.py:431] init engine (profile, create kv cache, warmup model) took 21.05 seconds
(ActorModelRayActor pid=1404977) Installed CUDA version 12.6 does not match the version torch was compiled with 12.4 but since the APIs are compatible, accepting this combination

sh文件：内容
`set -x
ray stop
CUDA_VISIBLE_DEVICES=0,1,2,3
NCCL_P2P_DISABLE=1
# reinforce++
ray start --head --node-ip-address 0.0.0.0 --num-gpus 4
sleep 5  # 等待10秒
ray status

ray job submit --address="http://127.0.0.1:8265" \
   -- python3 -m openrlhf.cli.train_ppo_ray \
   --ref_num_nodes 1 \
   --ref_num_gpus_per_node 1 \
   --actor_num_nodes 1 \
   --actor_num_gpus_per_node 2 \
   --vllm_num_engines 1 \
   --vllm_tensor_parallel_size 1 \
   --pretrain /home/huggingface_down/Qwen/Qwen2.5-1.5B-Instruct \
   --remote_rm_url /home/Effecienty_r1/OpenRLHF/openrlhf/rewards/reward_func.py \
   --save_path /homeEffecienty_r1/OpenRLHF/AAA_saves/rf_test/ \
   --micro_train_batch_size 16 \
   --train_batch_size 128 \
   --micro_rollout_batch_size 32 \
   --rollout_batch_size 256 \
   --n_samples_per_prompt 8 \
   --max_epochs 1 \
   --prompt_max_len 1024 \
   --max_samples 100000 \
   --generate_max_len 1024 \
   --init_kl_coef 1e-3 \
   --gamma 1.0 \
   --use_kl_loss \
   --kl_estimator k3 \
   --advantage_estimator group_norm \
   --zero_stage 2 \
   --bf16 \
   --actor_learning_rate 5e-7 \
   --prompt_data /home/Effecienty_r1/OpenRLHF/AAA_short_dataset_10k \
   --input_key context_messages \
   --apply_chat_template \
   --normalize_reward \
   --adam_offload \
   --gradient_checkpointing \
   --packing_samples \
   --save_steps -1 \
   --ckpt_path /home/Effecienty_r1/OpenRLHF/AAA_saves/rf_test/checkpoint

# You could also try
#   --kl_estimator k2 \

# also supports --advantage_estimator rloo | reinforce_baseline

`

**Comment by HaoZhongkai** [comment]
请问你是卡在哪里了？是engine initialized process group那里吗

**Comment by BeSkyer** [comment]
> 请问你是卡在哪里了？是engine initialized process group那里吗

你好， 我会在上面提到的这个地方卡住:
(LLMRayActor pid=1419840) INFO 03-13 02:08:05 llm_engine.py:431] init engine (profile, create kv cache, warmup model) took 20.52 seconds
(ActorModelRayActor pid=1419836) Installed CUDA version 12.6 does not match the version torch was compiled with 12.4 but since the APIs are compatible, accepting this combination

**Comment by HaoZhongkai** [comment]
那和我的报错不一样。我的通过 https://github.com/OpenRLHF/OpenRLHF/issues/492 解决了。 我在init engine后就开始训练了

**Comment by TimelessXZY** [comment]
您好请问这里解决了吗，遇到了一模一样的问题，我们是单机双卡，也会在这里卡住

> 后面换成了ray start --head --node-ip-address 127.0.0.1 --port=8266 --dashboard-port=8265 --num-gpus 4
> 依然会卡住
> Capturing CUDA graph shapes: 66%|██████▌ | 23/35 [00:10<00:05, 2.27it/s]
> (LLMRayActor pid=1419840)
> Capturing CUDA graph shapes: 69%|██████▊ | 24/35 [00:10<00:04, 2.26it/s]
> (LLMRayActor pid=1419840)
> Capturing CUDA graph shapes: 71%|███████▏ | 25/35 [00:11<00:04, 2.27it/s]
> (LLMRayActor pid=1419840)
> Capturing CUDA graph shapes: 74%|███████▍ | 26/35 [00:11<00:04, 2.24it/s]
> (LLMRayActor pid=1419840)
> Capturing CUDA graph shapes: 77%|███████▋ | 27/35 [00:12<00:03, 2.25it/s]
> (LLMRayActor pid=1419840)
> Capturing CUDA graph shapes: 80%|████████ | 28/35 [00:12<00:03, 2.26it/s]
> (LLMRayActor pid=1419840)
> Capturing CUDA graph shapes: 83%|████████▎ | 29/35 [00:13<00:02, 2.26it/s]
> (LLMRayActor pid=1419840)
> Capturing CUDA graph shapes: 86%|████████▌ | 30/35 [00:13<00:02, 2.26it/s]
> (LLMRayActor pid=1419840)
> Capturing CUDA graph shapes: 89%|████████▊ | 31/35 [00:13<00:01, 2.26it/s]
> (LLMRayActor pid=1419840)
> Capturing CUDA graph shapes: 91%|█████████▏| 32/35 [00:14<00:01, 2.26it/s]
> (LLMRayActor pid=1419840)
> Capturing CUDA graph shapes: 94%|█████████▍| 33/35 [00:14<00:00, 2.27it/s]
> (LLMRayActor pid=1419840)
> Capturing CUDA graph shapes: 97%|█████████▋| 34/35 [00:15<00:00, 2.26it/s]
> (LLMRayActor pid=1419840)
> Capturing CUDA graph shapes: 100%|██████████| 35/35 [00:15<00:00, 2.24it/s]
> Capturing CUDA graph shapes: 100%|██████████| 35/35 [00:15<00:00, 2.22it/s]
> (LLMRayActor pid=1419840) INFO 03-13 02:08:05 model_runner.py:1562] Graph capturing finished in 16 secs, took 0.20 GiB
> (LLMRayActor pid=1419840) INFO 03-13 02:08:05 llm_engine.py:431] init engine (profile, create kv cache, warmup model) took 20.52 seconds
> (ActorModelRayActor pid=1419836) Installed CUDA version 12.6 does not match the version torch was compiled with 12.4 but since the APIs are compatible, accepting this combination
